src/probnum/linops/_arithmetic.py

```diff
@@ -65,11 +391,21 @@ def _apply(
     fallback_operator: Optional[
         Callable[
             [LinearOperator, LinearOperator],
-            Union[LinearOperator, type(NotImplemented)],
+            Union[LinearOperator, NotImplementedType],
         ]
     ] = None,
-) -> Union[LinearOperator, type(NotImplemented)]:
-    key = (type(op1), type(op2))
+) -> Union[LinearOperator, NotImplementedType]:
+    if np.isscalar(op1):
+        key1 = "scalar"
```

Comment:
We should also normalize types to `numpy.number` here (`probnum.utils.as_numpy_scalar`). This makes things easier downstream. In this case, I think we don't really need the `"scalar"` key anymore.